automation/gtm_automation.js

```diff
@@ -0,0 +1,496 @@
+  limitations under the License.
+*/
+
+// The OAuth token client used to authenticate API calls.
+let tokenClient;
+
+// the GTM variables we will create. This is augmented with 'attribution' if
+// the attribution build is used.
+const VariableNames = [
+  'name',
+  'id',
+  'rating',
+  'value',
+  'delta',
+];
+
+/**
+ * Disables all of the form elements to avoid people making changes while the
+ * script runs.
+ */
+function lockForm() {
+  const inputs = Array.from(document.getElementsByTagName('input'));
+  inputs.map((e) => e.disabled = true);
+}
+
+/**
+ * Enables all of the form elements.
+ */
+function unlockForm() {
+  const inputs = Array.from(document.getElementsByTagName('input'));
+  inputs.map((e) => e.disabled = false);
+}
+
+/**
+ * Adds a message to the success-messages div on the page.
+ *
+ * @param {...string} message The message to add to the page. Arguments are joined
+ * with a space.
+ */
+function addSuccessMessage(...message) {
+  const successDiv = document.getElementById('success-messages');
+  successDiv.innerHTML += message.join(' ') + '<br>';
+}
+
+/**
+ * Adds a message to the error-messages div on the page.
+ *
+ * Also unlocks the form, since the script dies on an error.
+ *
+ * @param {...string} message The error message to add to the page. Arguments
+ * are joined with a space.
+ */
+function addErrorMessage(...message) {
+  const errorDiv = document.getElementById('error-messages');
+  errorDiv.innerHTML += message.join(' ') + '<br>';
+  unlockForm();
+}
+
+/**
+ * Runs the authorization flow to allow the app to use the GTM API. If completed
+ * successfully, the deployment of the template, tag, etc. is started.
+ *
+ * The user will be asked in a pop-up to choose an account to authorise with,
+ * and then asked to allow the app to edit GTM containers.
+ *
+ * @param {Event} event The event that triggers the function. This event is
+ * ignored.
+ */
+function authorizeApp(event) {
+  event.preventDefault();
+  const clientId = document.getElementById('oauth-id').value;
+  tokenClient = google.accounts.oauth2.initTokenClient({
+    client_id: clientId,
+    scope: 'https://www.googleapis.com/auth/tagmanager.edit.containers',
+    callback: '',
+  });
+  gapi.load('client', () => {
+    gapi.client.init({})
+      .then(function() {
+        gapi.client.load('https://tagmanager.googleapis.com/$discovery/rest?version=v2');
+      })
+      .then(runDeployment());
+  });
+}
+
+/**
+ * Start point for deploying the template, tags, etc. to GTM.
+ *
+ * The configuration is collected from the web interface here to centralize
+ * where the document is accessed. This config is then passed down the chain as
+ * the GTM API calls are made.
+ *
+ * The template file is read in here and then passed to createTemplate() along
+ * with the config to start the deployment workflow.
+ */
+function runDeployment() {
+  lockForm();
+  // set up the global config
+  const gtmURL = document.getElementById('gtm-url').value;
```

Comment:
nit: Should be gtmUrl
See https://google.github.io/styleguide/tsguide.html#camel-case
```suggestion
  const gtmUrl = document.getElementById('gtm-url').value;
```